units/task_monitor_execution_timeout.go

```diff
@@ -142,7 +143,11 @@ func cleanUpTimedOutTask(t task.Task) error {
 	// try to reset the task
 	taskID := t.Id
 	if t.IsPartOfDisplay() {
+		if displayTasksRestarted[taskID] {
+			return nil // already restarted this display task this run
+		}
 		taskID = t.DisplayTask.Id
+		displayTasksRestarted[taskID] = true
```

Comment:
Changing state of arguments is usually considered as an antipattern. I'd rather would use return value.